plugins/couchbase/couchbaseembed.cpp

```diff
@@ -360,6 +366,323 @@ namespace couchbaseembed
+                            {
+                                DBGLOG("Couchbase: Released connection object %p", connectionPtr);
+                            }
+                        }
+                    }
+                }
+            }
+
+            void expire()
+            {
+                if (!idleConnections.empty())
+                {
+                    CriticalBlock block(cacheLock);
+
+                    time_t oldestAllowedTime = time(NULL) - OBJECT_EXPIRE_TIMEOUT_SECONDS;
+                    __int32 expireCount = 0;
+
+                    for (ObjMap::iterator keyIter = idleConnections.begin(); keyIter != idleConnections.end(); keyIter++)
+                    {
+                        ConnectionList::iterator connectionIter = keyIter->second.begin();
+
+                        while (connectionIter != keyIter->second.end())
+                        {
+                            if (*connectionIter)
+                            {
+                                if ((*connectionIter)->getTimeTouched() < oldestAllowedTime)
+                                {
+                                    delete(*connectionIter);
+                                    connectionIter =  keyIter->second.erase(connectionIter);
+                                    ++expireCount;
+                                }
+                                else
+                                {
+                                    ++connectionIter;
+                                }
+                            }
+                            else
+                            {
+                                connectionIter =  keyIter->second.erase(connectionIter);
+                            }
+                        }
+                    }
+
+                    if (traceLevel > 4 && expireCount > 0)
+                    {
+                        DBGLOG("Couchbase: Expired %d cached connection%s", expireCount, (expireCount == 1 ? "" : "s"));
+                    }
+                }
+            }
+
+            CouchbaseConnection* getConnection(bool useSSL, const char * host, unsigned port, const char * bucketname, const char * password, const char * connOptions, unsigned int maxConnections)
+            {
+                CouchbaseConnection* connectionObjPtr = NULL;
+                StringBuffer connectionString = CouchbaseConnection::makeConnectionString(useSSL, host, port, bucketname, connOptions);
+
+                // Use a hash of the connection string as the key to finding
+                // any idle connection objects
+                hash64_t key = rtlHash64VStr(connectionString.str(), 0);
+
+                while (true)
+                {
+                    {
+                        CriticalBlock block(cacheLock);
+                        ConnectionList& idleConnectionList = idleConnections[key];
+                        bool useOnlyCache = maxConnections > 0 && activeConnections[key].size() >= maxConnections;
```

Comment:
I had included that boolean as a clarification of intention.  I've changed the logic and moved the expression into the if() statement that uses it.